engine/src/main/scala/cromwell/webservice/MetadataBuilderActor.scala

```diff
@@ -128,41 +128,60 @@ object MetadataBuilderActor {
       case s => Option(s.toInt)
     }
   }
+
+  private case object IndexedEmptyJson extends IndexedJsonValue {
+    override def index: Option[Int] = None
+    override def withIndex(index: String): IndexedJsonValue = this
+    override def toJson: JsValue = throw new IllegalStateException("IndexedEmptyJson cannot be converted to Json")
+  }
```

Comment:
I remember thinking a while back that since this stuff has a natural zero that perhaps one could combine monoid and foldable and replace some of the appends + such w/ folds. I think it's changed a fair amount since then though so YMMV